Libraries/LibJS/Runtime/StringPrototype.cpp

```diff
@@ -425,4 +426,37 @@ Value StringPrototype::slice(Interpreter& interpreter)
     return js_string(interpreter, string_part);
 }
 
+Value StringPrototype::last_index_of(Interpreter& interpreter)
+{
+    auto string = string_from(interpreter);
+    if (string.is_null())
+        return {};
+
+    if (interpreter.argument_count() == 0)
+        return Value(-1);
+
+    auto search_string = interpreter.argument(0).to_string();
+    if (search_string.length() > string.length())
+        return Value(-1);
+
+    ssize_t max_index = string.length() - search_string.length();
+    ssize_t from_index = max_index;
+    if (interpreter.argument_count() >= 2) {
+        from_index = static_cast<ssize_t>(interpreter.argument(1).to_i32());
+
+        if (from_index < 0)
+            from_index = 0;
+        if (from_index > max_index)
+            from_index = max_index;
+    }
+
+    for (ssize_t i = from_index; i >= 0; --i) {
+        auto part_view = string.substring_view(i, search_string.length());
+        if (part_view == search_string)
+            return Value((i32)i);
+    }
```

Comment:
Definitely not the fastest backward string search algorithm in the world, but I suppose it will be ok for now. It's always nice to leave performance opportunities for our future selves to find :)

Reply:
Not really, but that's what I was able to think for now. :)